sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_patch.py

```diff
@@ -36,3 +36,27 @@ def _validate_text_records(records):
         else:
             request_batch.append(doc)
     return request_batch
+
+def _get_positional_body(*args, **kwargs):
+    """Verify args and kwargs are valid, and then return the positional body, if users passed it in."""
+    if len(args) > 1:
+        raise TypeError("There can only be one positional argument, which is the POST body of this request.")
+    if args and "options" in kwargs:
+        raise TypeError(
+            "You have already supplied the request body as a positional parameter, "
+            "you can not supply it as a keyword argument as well."
+        )
+    return args[0] if args else None
+
+def _verify_qna_id_and_question(query_knowledgebase_options):
+    """For query_knowledge_base we require either `question` or `qna_id`."""
+    try:
+        qna_id = query_knowledgebase_options.qna_id
+        question = query_knowledgebase_options.question
+    except AttributeError:
+        qna_id = query_knowledgebase_options.get("qna_id") or query_knowledgebase_options.get("qnaId")
+        question = query_knowledgebase_options.get("question")
+    if not (qna_id or question):
```

Comment:
Should you also check that it's one or the other, not both?

Reply:
ahh good point